[Lens] Integrate Config Builder into APIs and transforms

nickofthyme opened the pull request:
## Summary

Lens transforms Phase 2

### Changes

#### Integrate the Config Builder

This integrates the `LensConfigBuilder` into the full end-to-end flow of all aspects of Lens including:
- Lens APIs (transforms config to pass over the wire, convert to SO for saving)
- Lens public client (transform api format to old Lens state)
- Lens transforms (transforms Lens by-value panels for dashboards)

#### Update Lens CM and API schema

The schemas following changes from #229534, did not match the integration perfectly and stubbed out a few things. So there was some cleanup needed to these schema to make them clearer and easier to understand.

These also needed to be changed to integrate the new `lensApiStateSchema` that will include all New API Formats.

#### Registers Lens embeddable transforms

Transforms:
- `dashboard/server` - Used to transform by-ref Lens dashboard panels.
- `dashboard/public` - Used to transform panels from url state.
- `lens/public` - Used to transform standalone Lens embeddables (see example [here](https://github.com/elastic/kibana/blob/ed9f709ef56b182116ebcd788245fdecfdacfd8b/x-pack/platform/plugins/shared/cases/public/components/visualizations/lens_renderer.tsx#L40)).

These transforms are meant to transform both:
- **Item transforms** - Current `v0` Lens state/attributes with old configurations to the latest state/attributes. Applying all existing runtime migrations and pins version to latest Lens item `v1`.
  - Only for `lens/public` transforms, we are not saving standalone Lens embeddable state in new API format.
- **API Format transforms** - Transforms the current `v1` Lens state to the new API format and vis-versa.
  - Needed to save Lens as `v1` as SO but pass the New API Format over the wire.
  - Needed to save Lens state in dashboard by-value panels in the New API Format.

Transforms are behind a features flag, when enabled it will transform only supported chart types for all lens visualizations including by-ref and by-value. When disabled (default) it will pass around the only Lens SO state as it does today.

```yml
# kibana.dev.yml
feature_flags.overrides:
  lens.apiFormat: true
```

### Issues

Closes #229543
Closes #229546
Closed #237192

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)

nickofthyme committed:
cleanup api/cm schema and types

nickofthyme committed:
integrate config builder into lens api and clients

nickofthyme committed:
add public and server lens embeddable transforms

nickofthyme committed:
add embeddable initial state transforms

This is needed in case of dashboard url state that would bypass other transforms for by-value lens panels.

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

nickofthyme committed:
dataview fix from peter

ppisljar committed:
Merge branch 'main' into lens-cb-integration

nickofthyme committed:
Merge branch 'main' into lens-cb-integration

nickofthyme committed:
Merge remote-tracking branch 'origin/lens-cb-integration' into lens-cb-integration

dej611 committed:
:sparkles: Add id to the schema

dej611 committed:
:bug: Fix the background_chart and dataViews bugs

nickofthyme committed:
Merge remote-tracking branch 'origin/lens-cb-integration' into lens-cb-integration

dej611 committed:
Merge remote-tracking branch 'upstream/main' into lens-cb-integration

nickofthyme committed:
Merge remote-tracking branch 'origin/lens-cb-integration' into lens-cb-integration

nickofthyme committed:
Merge branch 'main' into lens-cb-integration

nickofthyme committed:
update serialized types

nickofthyme committed:
add transforms for by-value panels

nickofthyme committed:
update lens embeddable to use/transform api format

nickofthyme committed:
feat: allow create with custom ids

nickofthyme committed:
fix: lens by-value panel refs

nickofthyme committed:
add feature flag to enable api transforms

nickofthyme committed:
Merge branch 'main' into lens-cb-integration

nickofthyme committed:
fix type error with response item

This item should never return the old `v0` state of lens.

nickofthyme committed:
fix custom lens renderer, add parentApi types

nickofthyme committed:
fix api attributes types

nickofthyme committed:
Merge branch 'main' into lens-cb-integration

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/346901)
* Commit: 7563388707324bf868c006244359278654c678da

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/346901#0199a60b-cc4c-4733-9e9b-fd50f6026e88)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346901","number":346901,"commit":"7563388707324bf868c006244359278654c678da"}],"number":346901}
buildkite-pr-comment-->

nickofthyme commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/346913)
* Commit: 2b5bab1edc79fc3381bfb030ad5703d572ae40ea

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/346913#0199a62f-9aa3-4692-a55d-9529ed677630)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346913","number":346913,"commit":"2b5bab1edc79fc3381bfb030ad5703d572ae40ea"}],"number":346913}
buildkite-pr-comment-->

nickofthyme committed:
[Lens] move references to server

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
chore: cleanup transforms, handle dynamic actions, stub out api config transforms

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
fix type errors

nickofthyme committed:
Merge branch 'main' into lens-server-refs

dej611 committed:
Merge branch 'main' into lens-server-refs

dej611 committed:
Merge remote-tracking branch 'upstream/main' into lens-server-refs

dej611 committed:
:label: first type pass

dej611 committed:
:label: More types fixed

dej611 committed:
:wrench: Flag package for treeshake

dej611 committed:
:white_check_mark: Align test to lack of references

nickofthyme committed:
chore: remove `schema` on transform definition

nickofthyme committed:
chore: move `LensParentApi` types to `@kbn/lens-common`

nickofthyme committed:
chore: revert code cleanup

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
fix: lens api integration tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

dej611 committed:
Merge branch 'lens-server-refs' into lens-cb-integration

dej611 committed:
:wrench: fix more type issues

dej611 committed:
:fire: Remvoe unused dep

nickofthyme committed:
:boom: untangle type explosion

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

nickofthyme committed:
fix add panel from library

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
Merge remote-tracking branch 'origin/lens-server-refs' into lens-server-refs

nickofthyme committed:
fix: background session issue

bg sessions use the public url transforms which defaults to references from the dashboard url which are now always empty, soon to be removed entirely.

But in Lens we had logic to filter references from the dashboard based on references from the lens SO. Instead we should just merge all refs from the lens SO and dashboard.

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
chore: update cloud limits

nickofthyme committed:
Merge remote-tracking branch 'origin/lens-server-refs' into lens-server-refs

nickofthyme committed:
Merge branch 'main' into lens-server-refs

nickofthyme committed:
Merge branch 'lens-server-refs' into lens-cb-integration

nickofthyme committed:
fix: by-value loading

nickofthyme committed:
Merge branch 'main' into lens-cb-integration

nickofthyme committed:
fix: numerous by-value issues with refs and transforms

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/357367)
* Commit: da9200983080aec62491a50a36e4da238fb227ec

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/357367#019a3c06-d066-4366-8de4-4c6b5208ce45)

### History
* :broken_heart: [Build #355985](https://buildkite.com/elastic/kibana-pull-request/builds/355985) failed b8317e0326eca365c489082e170920fe6b3a7381
* :broken_heart: [Build #346913](https://buildkite.com/elastic/kibana-pull-request/builds/346913) failed 2b5bab1edc79fc3381bfb030ad5703d572ae40ea

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357367","number":357367,"commit":"da9200983080aec62491a50a36e4da238fb227ec"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355985","number":355985,"commit":"b8317e0326eca365c489082e170920fe6b3a7381"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346913","number":346913,"commit":"2b5bab1edc79fc3381bfb030ad5703d572ae40ea"}],"number":357367}
buildkite-pr-comment-->

nickofthyme committed:
test: temporary type check fix

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/357876)
* Commit: 8dbf4e528ed0a91ec5b8d7bc69e4c3a527a473d9

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/357876#019a4b5d-6422-4a9d-9000-5a7e36a59387)

### History
* :broken_heart: [Build #357367](https://buildkite.com/elastic/kibana-pull-request/builds/357367) failed da9200983080aec62491a50a36e4da238fb227ec
* :broken_heart: [Build #355985](https://buildkite.com/elastic/kibana-pull-request/builds/355985) failed b8317e0326eca365c489082e170920fe6b3a7381
* :broken_heart: [Build #346913](https://buildkite.com/elastic/kibana-pull-request/builds/346913) failed 2b5bab1edc79fc3381bfb030ad5703d572ae40ea

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357876","number":357876,"commit":"8dbf4e528ed0a91ec5b8d7bc69e4c3a527a473d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357367","number":357367,"commit":"da9200983080aec62491a50a36e4da238fb227ec"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355985","number":355985,"commit":"b8317e0326eca365c489082e170920fe6b3a7381"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346913","number":346913,"commit":"2b5bab1edc79fc3381bfb030ad5703d572ae40ea"}],"number":357876}
buildkite-pr-comment-->

nickofthyme committed:
temporary reset limits

nickofthyme commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/357916)
* Commit: e1e608252f175597ed5571495b8b3c88240f027d

### Failed CI Steps
* [Jest Integration Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/357916#019a4c15-da88-4b04-8ec0-3701958ab85d)
* [FTR Configs #12](https://buildkite.com/elastic/kibana-pull-request/builds/357916#019a4c15-da9d-47e8-b70b-73c83e6dff28)
* [FTR Configs #12](https://buildkite.com/elastic/kibana-pull-request/builds/357916#019a4c3c-ae0c-43be-a64a-89415aad09e9)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/357916#019a4c15-da88-4b04-8ec0-3701958ab85d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/357916/jobs/019a4c15-da88-4b04-8ec0-3701958ab85d/artifacts/019a4c32-d6a3-47a2-adc8-3040c14bdc12) Jest Integration Tests #<span></span>8 / daily rollups integration test deletes documents older that 3 days from the saved objects repository
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/357916#019a4c15-da9d-47e8-b70b-73c83e6dff28) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/357916/jobs/019a4c15-da9d-47e8-b70b-73c83e6dff28/artifacts/019a4c3c-96b7-4aef-8c5b-8fdf4663e0da) FTR Configs #<span></span>12 / lens visualizations visualizations - create validation should return error if body is empty
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/357916#019a4c3c-ae0c-43be-a64a-89415aad09e9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/357916/jobs/019a4c3c-ae0c-43be-a64a-89415aad09e9/artifacts/019a4c42-2824-4251-8080-5b67b401286a) FTR Configs #<span></span>12 / lens visualizations visualizations - create validation should return error if body is empty

### History
* :broken_heart: [Build #357876](https://buildkite.com/elastic/kibana-pull-request/builds/357876) failed 8dbf4e528ed0a91ec5b8d7bc69e4c3a527a473d9
* :broken_heart: [Build #357367](https://buildkite.com/elastic/kibana-pull-request/builds/357367) failed da9200983080aec62491a50a36e4da238fb227ec
* :broken_heart: [Build #355985](https://buildkite.com/elastic/kibana-pull-request/builds/355985) failed b8317e0326eca365c489082e170920fe6b3a7381
* :broken_heart: [Build #346913](https://buildkite.com/elastic/kibana-pull-request/builds/346913) failed 2b5bab1edc79fc3381bfb030ad5703d572ae40ea

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357916","number":357916,"commit":"e1e608252f175597ed5571495b8b3c88240f027d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357876","number":357876,"commit":"8dbf4e528ed0a91ec5b8d7bc69e4c3a527a473d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/357367","number":357367,"commit":"da9200983080aec62491a50a36e4da238fb227ec"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355985","number":355985,"commit":"b8317e0326eca365c489082e170920fe6b3a7381"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/346913","number":346913,"commit":"2b5bab1edc79fc3381bfb030ad5703d572ae40ea"}],"number":357916}
buildkite-pr-comment-->

nickofthyme committed:
fix: ts cyclic type import.....why!!!!

nickofthyme committed:
fix: mounter type errors

nickofthyme committed:
fix: external attribute usage, add legacy state method

nickofthyme committed:
fix: remaining type errors

nickofthyme committed:
ci: update empty error responses for bad schema union messages

nickofthyme committed:
Merge branch 'main' into lens-cb-integration

kibanamachine committed:
Changes from node scripts/capture_oas_snapshot   --include-path /api/status   --include-path /api/alerting/rule/   --include-path /api/alerting/rules   --include-path /api/actions   --include-path /api/security/role   --include-path /api/spaces   --include-path /api/streams   --include-path /api/fleet   --include-path /api/saved_objects/_import   --include-path /api/saved_objects/_export   --include-path /api/maintenance_window   --include-path /api/agent_builder --update

nickofthyme committed:
refactor: lazily import builder in `lens/public`, reset limits

nickofthyme committed:
Merge branch 'main' into lens-cb-integration

nickofthyme committed:
Merge remote-tracking branch 'origin/lens-cb-integration' into lens-cb-integration

nickofthyme commented:
/ci

kibanamachine committed:
Changes from node scripts/generate codeowners

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/358657)
* Commit: 408e743454c14be17f4c1102b33ce94e85b8df0e

### Failed CI Steps
* [Jest Tests #21](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c)
* [Jest Tests #21](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e)
* [FTR Configs #34](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b4f-4748-a9c5-ead22bb01e5e)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9b7c-45f0-bd78-39ea9f3bbe19) Jest Tests #<span></span>21 / AlertRuleFromVisAction appends a single xValue to the threshold line with an AND operator
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-15d6-4f9a-8493-31b95d971648) Jest Tests #<span></span>21 / AlertRuleFromVisAction appends a single xValue to the threshold line with an AND operator
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9b7e-4615-bbe7-b5bf73c9dd52) Jest Tests #<span></span>21 / AlertRuleFromVisAction appends multiple fields in the threshold value with an AND operator
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-15d7-4504-b706-987acbcf1ee6) Jest Tests #<span></span>21 / AlertRuleFromVisAction appends multiple fields in the threshold value with an AND operator
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9b80-4963-9969-1f1a18549b5d) Jest Tests #<span></span>21 / AlertRuleFromVisAction appends multiple thresholdValues threshold line in parentheses separated by OR operators
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-15d8-49d9-ad38-97620518e8db) Jest Tests #<span></span>21 / AlertRuleFromVisAction appends multiple thresholdValues threshold line in parentheses separated by OR operators
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9b7d-4d18-a79d-e24b63869fab) Jest Tests #<span></span>21 / AlertRuleFromVisAction converts an array in a threshold value to MATCH queries
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-15d6-4280-a627-529ffad510a8) Jest Tests #<span></span>21 / AlertRuleFromVisAction converts an array in a threshold value to MATCH queries
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9b7f-4348-a15e-a70197f62379) Jest Tests #<span></span>21 / AlertRuleFromVisAction converts an array xValue to MATCH queries
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-15d7-4a7a-9bac-ccd8a7e054be) Jest Tests #<span></span>21 / AlertRuleFromVisAction converts an array xValue to MATCH queries
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9bc1-4a8c-8797-881acb37f507) Jest Tests #<span></span>21 / AlertRuleFromVisAction creates a rule with the visualization's ES|QL query plus an additional threshold line
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-165d-4b6c-85f5-5067604efe08) Jest Tests #<span></span>21 / AlertRuleFromVisAction creates a rule with the visualization's ES|QL query plus an additional threshold line
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9b7c-44a5-aa6d-cc00e47b4573) Jest Tests #<span></span>21 / AlertRuleFromVisAction does not duplicate function column renames when they are included in multiple threshold values
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-15d6-475a-92a4-4ca8917c4378) Jest Tests #<span></span>21 / AlertRuleFromVisAction does not duplicate function column renames when they are included in multiple threshold values
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9b7b-43ac-9a4f-72d151281a96) Jest Tests #<span></span>21 / AlertRuleFromVisAction escapes string values with backlashes in them
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-15d5-464d-97c2-26f7db9fa855) Jest Tests #<span></span>21 / AlertRuleFromVisAction escapes string values with backlashes in them
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9b80-4463-bc87-aebd9e1e9fd8) Jest Tests #<span></span>21 / AlertRuleFromVisAction escapes unnamed function columns
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-15d8-4238-bed5-6624b8278224) Jest Tests #<span></span>21 / AlertRuleFromVisAction escapes unnamed function columns
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9b7f-438f-8daa-83eec9c48141) Jest Tests #<span></span>21 / AlertRuleFromVisAction renders empty splitValues as empty strings
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-15d7-4738-bf2d-4cee4dc32e39) Jest Tests #<span></span>21 / AlertRuleFromVisAction renders empty splitValues as empty strings
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9bc2-42a0-8d79-263d668d4443) Jest Tests #<span></span>21 / AlertRuleFromVisAction when executed without a data parameter derives data from the embeddable and uses placeholder threshold values
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-165d-4a04-8cf7-e6c4e2176776) Jest Tests #<span></span>21 / AlertRuleFromVisAction when executed without a data parameter derives data from the embeddable and uses placeholder threshold values
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54ea-6b1f-46cf-9574-1128a0aafc8c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54ea-6b1f-46cf-9574-1128a0aafc8c/artifacts/019a54fd-9b80-4b40-a262-63c6fdc7468e) Jest Tests #<span></span>21 / AlertRuleFromVisAction when executed without a data parameter uses placeholder split values when the X axis is not a timestamp
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358657#019a54fd-b4ea-42bf-8056-227a6c9ce94e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358657/jobs/019a54fd-b4ea-42bf-8056-227a6c9ce94e/artifacts/019a5511-15d8-40c1-a65d-7bf97b2dff96) Jest Tests #<span></span>21 / AlertRuleFromVisAction when executed without a data parameter uses placeholder split values when the X axis is not a timestamp

### History


<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358657","number":358657,"commit":"408e743454c14be17f4c1102b33ce94e85b8df0e"}],"number":358657}
buildkite-pr-comment-->

nickofthyme committed:
fix: alert action tests

nickofthyme committed:
Merge remote-tracking branch 'origin/lens-cb-integration' into lens-cb-integration

elasticmachine commented:
Pinging @elastic/kibana-visualizations (Team:Visualizations)

nickofthyme committed:
Merge branch 'main' into lens-cb-integration

nickofthyme commented on src/platform/packages/shared/kbn-lens-common-2/index.ts:
All these types were originally in `@kbn/lens-common` but due to the need for...

```ts
import type { LensApiSchemaType } from '@kbn/lens-embeddable-utils';
```

This creates cyclical import path because `@kbn/lens-embeddable-utils` depends on `@kbn/lens-common`.

markov00 commented on src/platform/packages/shared/kbn-lens-common-2/README.md:
```diff
@@ -0,0 +1,7 @@
+# @kbn/lens-common-2
```
I hope this name is just temporary to have this working

markov00 commented on x-pack/platform/plugins/shared/lens/public/plugin.ts:
```diff
@@ -386,12 +388,21 @@ export class LensPlugin {
+            transformEnhancementsOut: embeddable.transformEnhancementsOut,
+          }).transformOut;
+        });
       });
```
Just wondering: isn't better to configure this in the start method instead of the setup?
If I'm not wrong the setup is a sync method. or probably do the opposite: get the core start services within the registerLegacy URL async function?

markov00 committed:
Merge branch 'main' into lens-cb-integration

nickofthyme commented on x-pack/platform/plugins/shared/lens/common/transforms/utils.ts:
```diff
@@ -21,3 +22,9 @@ export function findLensReference(references?: Reference[]) {
+export function isByRefLensConfig(
+  config: LensSerializedAPIConfig
+): config is LensByRefSerializedAPIConfig {
+  return !config.attributes;
```
Yeah possibly, or check both. I don't recall if there was a reason for doing this. But also the old logic was passing the `attributes` back even when saving a by-ref Lens panel. So I fixed that.

I'll update it to check the `savedObjectId`.

markov00 replied:
sounds good, thanks

nickofthyme commented on src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/charts/metric.ts:
```diff
@@ -181,6 +181,7 @@ const metricStateBreakdownByOptionsSchema = schema.object({
 });
 
 export const metricStateSchemaNoESQL = schema.object({
+  id: schema.maybe(schema.string()),
```
We don't, I think this is remnant of the multiple merges 🙃

nickofthyme committed:
chore: rename embeddable transform helpers

nickofthyme committed:
chore: cleanup naming

nickofthyme committed:
chore: remove old metric schema `id`

markov00 reviewed (approved):
Reviewed after accepting my previous comments replies. Thanks Nick, all good for me!

markov00 committed:
Merge branch 'main' into lens-cb-integration

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/359677)
* Commit: 72a1d30859438ff003db96f9c79d1992952ccab5

### Failed CI Steps
* [Jest Integration Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/359677#019a6d15-2b11-46bd-b8e7-f4c9ea7edfbd)
* [Jest Integration Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/359677#019a6d19-441a-4a86-bf61-d61fbb63faf7)

### History
* :green_heart: [Build #359162](https://buildkite.com/elastic/kibana-pull-request/builds/359162) succeeded 933f246ef1ccbb5992882e20ca093142d1196c47
* :yellow_heart: [Build #358905](https://buildkite.com/elastic/kibana-pull-request/builds/358905) was flaky 5847aca23064f8db47b98f3ecf0004fc82d04347
* :yellow_heart: [Build #358719](https://buildkite.com/elastic/kibana-pull-request/builds/358719) was flaky 4a806e3390e1c170d9c5d6a631b7c30698b1f629

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359677","number":359677,"commit":"72a1d30859438ff003db96f9c79d1992952ccab5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/359162","number":359162,"commit":"933f246ef1ccbb5992882e20ca093142d1196c47"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358905","number":358905,"commit":"5847aca23064f8db47b98f3ecf0004fc82d04347"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358719","number":358719,"commit":"4a806e3390e1c170d9c5d6a631b7c30698b1f629"}],"number":359677}
buildkite-pr-comment-->

dej611 commented on src/platform/packages/shared/kbn-lens-common-2/README.md:
Would it be best to include here all the Lens `Embeddable` types at this point?
I mean, generic Lens types at this point in `@kbn/lens-common` and embeddable types in `@kbn/lens-embeddable-common`. wdyt?

dej611 commented on x-pack/platform/plugins/shared/lens/common/feature_flags.ts:
```diff
@@ -0,0 +1,86 @@
+}
+
+function fetchFeatureFlagFn(service: FeatureFlagsStartPublic | FeatureFlagsStartServer) {
+  return async function fetchFeatureFlag(flag: LensFeatureFlag) {
```
nit: does it need to be `async`?
```suggestion
  return function fetchFeatureFlag(flag: LensFeatureFlag) {
```

nickofthyme replied:
Sure, updated in 5ca784463fae13a9527eb80e3a14cee2f1c712f1

dej611 commented on x-pack/platform/plugins/shared/lens/public/react_embeddable/lens_embeddable.tsx:
```diff
@@ -53,7 +54,7 @@ export const createLensEmbeddableFactory = (
      * @returns an object with the Lens API and the React component to render in the Embeddable
      */
-    buildEmbeddable: async ({ initialState, finalizeApi, parentApi, uuid }) => {
+    buildEmbeddable: async ({ finalizeApi, parentApi, uuid, initialState }) => {
```
why this change?

dej611 reviewed (approved):
General logic seems ok.
I'm a bit concerned about the `lens-common-2` package: perhaps it can have a better name & more organized content? Not a blocker tho.

nickofthyme committed:
chore: simplify async logic

nickofthyme committed:
chore: revert destructure order change

nickofthyme committed:
test: disable message check on empty response test

nickofthyme committed:
Merge branch 'main' into lens-cb-integration

nickofthyme committed:
chore: improve async logic in plugin
